pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -148,9 +176,16 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 	}
 
 	nextIndex := (b.firstIndex + b.size) % len(b.points)
-	b.points[nextIndex] = p
 	b.size++
-	return nil
+	return &b.points[nextIndex], nil
+}
+
+func (b *HPointRingBuffer) RemoveLastPoint() {
+	if b.size == 0 {
+		panic("There are no points to remove")
+	}
+
+	b.size--
```

Comment:
If `b.size` is 0 after decrementing, we should set `b.firstIndex` to 0 - this is consistent with what `Reset` and `DiscardPointsBefore` do, and is done to try to avoid wraparound if possible.